morpheus/service/vdb/milvus_vector_db_service.py

```diff
@@ -217,6 +384,158 @@ async def similarity_search(self,
+        **kwargs :  dict[str, typing.Any]
+            Extra keyword arguments specific to the vector database implementation.
+
+        Returns
+        -------
+        typing.Any
+            Returns result of the given keys that are deleted from the collection.
+        """
+
+        result = self._client.delete(collection_name=self._name, pks=keys, **kwargs)
+
+        self._collection.load()
+
+        return result
+
+    def delete(self, expr: str, **kwargs: dict[str, typing.Any]) -> dict[str, typing.Any]:
+        """
+        Delete vectors from the collection using expressions.
+
+        Parameters
+        ----------
+        expr : str
+            Delete expression.
+        **kwargs :  dict[str, typing.Any]
+            Extra keyword arguments specific to the vector database implementation.
+
+        Returns
+        -------
+        dict[str, typing.Any]
+            Returns result of the given keys that are deleted from the collection.
+        """
+
+        result = self._client.delete_by_expr(collection_name=self._name, expression=expr, **kwargs)
+
+        return self._update_delete_result_to_dict(result=result)
+
+    def retrieve_by_keys(self, keys: int | str | list, **kwargs: dict[str, typing.Any]) -> list[typing.Any]:
+        """
+        Retrieve the inserted vectors using their primary keys.
+
+        Parameters
+        ----------
+        keys : int | str | list
+            Primary keys to get vectors for. Depending on pk_field type it can be int or str
+            or a list of either.
+        **kwargs :  dict[str, typing.Any]
+            Additional keyword arguments for the retrieval operation.
+
+        Returns
+        -------
+        list[typing.Any]
+            Returns result rows of the given keys from the collection.
+        """
+
+        result = None
+
+        try:
+            result = self._client.get(collection_name=self._name, ids=keys, **kwargs)
+        except pymilvus.exceptions.MilvusException as exec_info:
+            raise RuntimeError(f"Unable to perform search: {exec_info}") from exec_info
+
+        return result
+
+    def count(self, **kwargs: dict[str, typing.Any]) -> int:
+        """
+        Returns number of rows/entities.
+
+        Parameters
+        ----------
+        **kwargs :  dict[str, typing.Any]
+            Additional keyword arguments for the count operation.
+
+        Returns
+        -------
+        int
+            Returns number of entities in the collection.
+        """
+        return self._collection.num_entities
+
+    def drop(self, **kwargs: dict[str, typing.Any]) -> None:
+        """
+        Drop a collection, index, or partition in the Milvus vector database.
+
+        This function allows you to drop a collection.
+
+        Parameters
+        ----------
+        **kwargs : dict
+            Additional keyword arguments for specifying the type and partition name (if applicable).
+        """
+
+        self._collection.drop(**kwargs)
```

Comment:
Does this require an associated flush?

Reply:
Drop doesn't need flush, as it is deleting the collection.